src/kili/queries/asset/helpers.py

```diff
@@ -73,30 +107,75 @@ def download_file(url: str, external_id: str, local_dir_path: Path) -> Path:
                 f"The asset does not have the {field} field. Please add it to the fields when"
-                " querying assets "
-            )
-        content = asset["content"]
-        if content == "":
-            raise NotImplementedError(
-                "The media of the asset is not stored in the content field. If your asset is video"
-                " imported from frames or a richText asset, the media is stored in the jsonContent"
-                " field. The download in local of such asset media is currently not supported."
+                " querying assets."
             )
 
 
+# pylint: disable=too-many-branches
 def download_asset_content(assets: List[dict], local_dir_path: Path) -> List[dict]:
     """Download assets media in local."""
     if len(assets) == 0:
         return assets
+    map(assert_required_fields_existence, assets)
+
+    asset_urls = []
+    external_ids = []
+    is_video_frame = []
+    for asset in assets:
+        if asset["content"] != "":
+            asset_urls.append(asset["content"])
+            external_ids.append(asset["externalId"])
+            is_video_frame.append(False)
+        elif asset["jsonContent"] != "":
+            if asset["project"]["inputType"] == "TEXT":
+                asset_urls.append(asset["jsonContent"])
+                external_ids.append(asset["externalId"])
+                is_video_frame.append(False)
+            elif asset["project"]["inputType"] == "VIDEO":
+                urls = get_json_content_urls(asset["jsonContent"])
+                asset_urls.extend(urls)
+                external_ids.extend([asset["externalId"]] * len(urls))
+                is_video_frame.extend([True] * len(urls))
+            else:
+                raise NotImplementedError(
+                    f'jsonContent download for type {asset["project"]["inputType"]} not implemented'
+                    " yet."
+                )
+
     local_dir_path.mkdir(parents=True, exist_ok=True)
-    assert_required_fields_existence(assets[0])
-    content_gen = (asset["content"] for asset in assets)
-    external_id_gen = (asset["externalId"] for asset in assets)
     with ThreadPoolExecutor() as threads:
-        path_gen = threads.map(download_file, content_gen, external_id_gen, repeat(local_dir_path))
-    return [{**asset, "content": str(path)} for asset, path in zip(assets, path_gen)]
+        path_gen = threads.map(
+            download_file, asset_urls, external_ids, is_video_frame, repeat(local_dir_path)
+        )
+
+        paths = list(path_gen)
+
+    # put paths to files back into assets dicts
+    i = j = 0
+    while i < len(assets):
+        if assets[i]["content"] != "":
+            assets[i] = {**assets[i], "content": str(paths[j])}
+            j += 1
+        elif assets[i]["jsonContent"] != "":
+            if assets[i]["project"]["inputType"] == "VIDEO":
+                assets[i] = {**assets[i], "content": str(paths[j].parent)}  # root dir for frames
```

Comment:
I think that we can't really put the folder name in the content field. it is not an informative field, but rather a technical field storing the information for the app on the asset media.
But I admit that it is ambiguous for local asset paths. We need to discuss this tomorrow at the daily.